### ISSUE TITLE
💅 noUnusedImports ignores valid TSDoc Link

### ISSUE THREAD

**Issue by davatron5000** [biomejs/biome#3812]
### Environment information

```bash
CLI:
  Version:                      1.8.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.12.2"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.5.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Linter:
  JavaScript enabled:           true
  JSON enabled:                 true
  CSS enabled:                  false
  Recommended:                  true
  All:                          false
  Enabled rules:
  performance/noDelete
  suspicious/noCatchAssign
  suspicious/noUnsafeNegation
  complexity/useLiteralKeys
  style/useImportType
  complexity/noMultipleSpacesInRegularExpressionLiterals
  a11y/useValidLang
  complexity/noUselessEmptyExport
  suspicious/useNamespaceKeyword
  suspicious/useValidTypeof
  a11y/useValidAriaRole
  correctness/noConstantCondition
  a11y/useAriaActivedescendantWithTabindex
  suspicious/noAssignInExpressions
  style/useDefaultParameterLast
  complexity/noEmptyTypeParameters
  correctness/noConstructorReturn
  style/useSelfClosingElements
  suspicious/noDuplicateParameters
  style/useTemplate
  correctness/noUnusedLabels
  complexity/noUselessTernary
  correctness/noUnreachableSuper
  suspicious/noCompareNegZero
  correctness/noSwitchDeclarations
  a11y/noAutofocus
  correctness/noUnsafeOptionalChaining
  correctness/noConstAssign
  suspicious/noControlCharactersInRegex
  complexity/noUselessTypeConstraint
  style/noVar
  suspicious/noDoubleEquals
  suspicious/noRedundantUseStrict
  style/useLiteralEnumMembers
  suspicious/noGlobalIsNan
  suspicious/noEmptyInterface
  suspicious/noConstEnum
  suspicious/noMisleadingCharacterClass
  correctness/noPrecisionLoss
  suspicious/noRedeclare
  correctness/noStringCaseMismatch
  correctness/noSetterReturn
  correctness/noInvalidConstructorSuper
  suspicious/noImplicitAnyLet
  suspicious/noFallthroughSwitchClause
  suspicious/noUnsafeDeclarationMerging
  correctness/noUnreachable
  a11y/useKeyWithClickEvents
  suspicious/noDuplicateObjectKeys
  complexity/noUselessThisAlias
  complexity/noThisInStatic
  complexity/useOptionalChain
  correctness/noInnerDeclarations
  style/noParameterAssign
  suspicious/noDuplicateCase
  a11y/useValidAnchor
  complexity/useRegexLiterals
  correctness/noSelfAssign
  style/noUselessElse
  style/useShorthandFunctionType
  suspicious/noShadowRestrictedNames
  a11y/useMediaCaption
  complexity/noUselessLabel
  complexity/noUselessCatch
  correctness/noUnsafeFinally
  a11y/useAriaPropsForRole
  correctness/noNonoctalDecimalEscape
  style/useEnumInitializers
  a11y/useHtmlLang
  suspicious/noDuplicateTestHooks
  complexity/noStaticOnlyClass
  style/useWhile
  complexity/useArrowFunction
  style/useNumericLiterals
  a11y/noNoninteractiveTabindex
  complexity/useSimpleNumberKeys
  correctness/useYield
  a11y/noInteractiveElementToNoninteractiveRole
  suspicious/noImportAssign
  correctness/noUnnecessaryContinue
  suspicious/noApproximativeNumericConstant
  suspicious/noLabelVar
  suspicious/useDefaultSwitchClauseLast
  correctness/noGlobalObjectCalls
  a11y/useAltText
  correctness/noEmptyCharacterClassInRegex
  suspicious/noSuspiciousSemicolonInJsx
  suspicious/noSparseArray
  a11y/useIframeTitle
  complexity/noBannedTypes
  a11y/noSvgWithoutTitle
  suspicious/noPrototypeBuiltins
  style/useAsConstAssertion
  correctness/useJsxKeyInIterable
  style/useExportType
  complexity/noUselessLoneBlockStatements
  style/noArguments
  a11y/useValidAriaValues
  correctness/noVoidElementsWithChildren
  suspicious/noDebugger
  suspicious/noMisleadingInstantiator
  suspicious/noGlobalAssign
  suspicious/noCommentText
  suspicious/noDuplicateJsxProps
  suspicious/noThenProperty
  a11y/noPositiveTabindex
  correctness/noEmptyPattern
  complexity/noExcessiveNestedTestSuites
  security/noDangerouslySetInnerHtmlWithChildren
  a11y/useKeyWithMouseEvents
  suspicious/useGetterReturn
  suspicious/noExtraNonNullAssertion
  correctness/noRenderReturnValue
  correctness/useExhaustiveDependencies
  security/noGlobalEval
  style/noNonNullAssertion
  a11y/noRedundantRoles
  complexity/useFlatMap
  correctness/useIsNan
  style/useConst
  suspicious/noGlobalIsFinite
  suspicious/noSelfCompare
  suspicious/noAsyncPromiseExecutor
  security/noDangerouslySetInnerHtml
  style/useNodejsImportProtocol
  a11y/noDistractingElements
  suspicious/noArrayIndexKey
  complexity/noWith
  suspicious/noDuplicateClassMembers
  complexity/noExtraBooleanCast
  performance/noAccumulatingSpread
  a11y/useValidAriaProps
  a11y/noRedundantAlt
  correctness/noChildrenProp
  suspicious/noConfusingLabels
  suspicious/noConfusingVoidType
  suspicious/noFocusedTests
  a11y/useButtonType
  a11y/noAriaUnsupportedElements
  correctness/noFlatMapIdentity
  a11y/noBlankTarget
  a11y/useHeadingContent
  correctness/useValidForDirection
  correctness/noVoidTypeReturn
  correctness/noInvalidUseBeforeDeclaration
  a11y/noAriaHiddenOnFocusable
  a11y/useAnchorContent
  complexity/noUselessRename
  correctness/noInvalidNewBuiltin
  style/useNumberNamespace
  complexity/noUselessConstructor
  a11y/noAccessKey
  style/useExponentiationOperator
  style/noUnusedTemplateLiteral
  complexity/noUselessSwitchCase
  style/useSingleVarDeclarator
  suspicious/noExportsInTest
  a11y/noNoninteractiveElementToInteractiveRole
  style/noCommaOperator
  suspicious/useIsArray
  a11y/noHeaderScope
  complexity/noUselessFragments
  suspicious/noMisrefactoredShorthandAssign
  correctness/noUnusedImports
  suspicious/noClassAssign
  suspicious/noFunctionAssign

Workspace:
  Open Documents:               0
```

### Rule name

noUnusedImports

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=aQBtAHAAbwByAHQAIAB0AHkAcABlACAAewAgAGQAZQBiAG8AdQBuAGMAZQAgAH0AIABmAHIAbwBtACAAJwBoAHQAdABwAHMAOgAvAC8AZQBzAG0ALgBzAGgALwBsAG8AZABhAHMAaAAuAGQAZQBiAG8AdQBuAGMAZQBAADQALgAwAC4AOAAnAAoACgAvACoAKgAKACAAKgAgAHsAQABsAGkAbgBrACAAZABlAGIAbwB1AG4AYwBlACAAfAAgAHQAaABlACAARABlAGIAbwB1AG4AYwBlACAAbQBlAHQAaABvAGQAIAB9ACAAaQBzACAAYwBvAG8AbAAKACAAKgAvAA%3D%3D

### Expected result

Don't error when imports (particularly types) are used inside valid TSDoc `{@link ...}` tags.

- https://tsdoc.org/pages/tags/link/

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
Are you not able to import type in TSDoc directly?

**Comment by SirzBenjie** [comment]
This issue doesn't have any labels on it, which tells me that it may have been forgotten. However, it is one of the most annoying things to me.

> Are you not able to import type in TSDoc directly?

Yes, you can. But you shouldn't have to. Also vscode screws this up (par for the course, it really sucks at parsing jsdoc/tsdoc) and will show the link.
We are having to add these ignore tags everywhere. And they're annoying.
Also both eslint and typescript properly identify these as usages.

**Comment by DayKev** [comment]
Looks like the issue in this case is that the TSDoc comment wasn't attached to anything. As soon as I added `const x = 3;` below the TSDoc in the example it stops seeing the import as unused ([playground link](https://biomejs.dev/playground/?code=aQBtAHAAbwByAHQAIAB0AHkAcABlACAAewAgAGQAZQBiAG8AdQBuAGMAZQAgAH0AIABmAHIAbwBtACAAJwBoAHQAdABwAHMAOgAvAC8AZQBzAG0ALgBzAGgALwBsAG8AZABhAHMAaAAuAGQAZQBiAG8AdQBuAGMAZQBAADQALgAwAC4AOAAnAAoACgAvACoAKgAKACAAKgAgAHsAQABsAGkAbgBrACAAZABlAGIAbwB1AG4AYwBlACAAfAAgAHQAaABlACAARABlAGIAbwB1AG4AYwBlACAAbQBlAHQAaABvAGQAfQAgAGkAcwAgAGMAbwBvAGwACgAgACoALwAKAGMAbwBuAHMAdAAgAHgAIAA9ACAAMwA7AAoAYwBvAG4AcwBvAGwAZQAuAGwAbwBnACgAeAApAA%3D%3D&language=ts)).
At least that's the issue now in Biome `2.3.8` (more specifically commit [8de2774](https://github.com/biomejs/biome/tree/8de2774fb507a10e32ecf920bb5d0f801a9e869c), which is what the Playground is using as of this comment), dunno about `1.8.3` which is what the OP was using back when the issue was opened 14+ months ago.

While there isn't much use for floating TSDoc comments, if you're using something like [Typedoc](https://typedoc.org/) which renders TSDoc comments to a website and using the [`@packageDocumentation`](https://tsdoc.org/pages/tags/packagedocumentation/) tag, you might have imports used in that documentation block that you want parsed, and using `import type { ... } from "..."` gives you type/syntax checking whereas using `@import ...` from within the TSDoc comment does not.

Also, a more precise issue title is "`noUnusedImports` ignores TSDoc usage in floating comment blocks".
